.github/workflows/build.yml

```diff
@@ -0,0 +1,58 @@
+name: build
+run-name: Build Pipeline
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_call:
+jobs:
+  build:
+    name: Verify, Build and Test
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+        python:
+          - "3.11"
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Display system and Python versions
+        run: |
+          python -c 'import platform as p; print(f"{p.python_implementation()} {p.python_version()} | {p.system()} {p.release()}")'
```

Comment:
is `python --version` not safe enough?

Reply:
Just wanted to add this as an FYI to track both the python and os versions in case it's relevant for debugging

Reply:
fair enough, but this does exist: https://github.com/microsoft/sammo/actions/runs/7560893210/job/20587901480?pr=11#step:1:2 :)

Reply:
oh that's a good point - I missed that. This step is probably redundant, then